core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -522,6 +527,15 @@ class PartKeyLuceneIndex(ref: DatasetRef,
     collector.result
   }
 
+  def labelNamesFromFilters(columnFilters: Seq[ColumnFilter],
+                          startTime: Long,
+                          endTime: Long,
+                          limit: Int): Seq[Int] = {
+    val partIdsOrderedByEndTime = new TopKPartIdsCollector(limit)
+    searchFromFilters(columnFilters, startTime, endTime, partIdsOrderedByEndTime)
+    partIdsOrderedByEndTime.topKPartIDsBitmap().toArray
```

Comment:
Top-k is an expensive query. Can we do a simple search and get 1 result ?

Reply:
Done 👍 , updated code to use SinglePartIdCollector